src/models/tests/gamma_poisson_model_test.py

```diff
@@ -218,6 +218,19 @@ def test_overspend(self, mock_gamma_poisson_model):
         self.assertAlmostEqual(gpm.by_impressions([30000]).reach(1), 10000.0, delta=0.1)
         self.assertAlmostEqual(gpm.by_spend([300]).reach(1), 10000.0, delta=0.1)
 
+    def test_fit_histogram_chi2_distance_with_extrapolation(self):
+        # The following point corresponds to
+        #  I, Imax, N, alpha0, beta0 = 5000, 30000, 10000, 1.0, 2.0
+        # This represents a distribution with mean = 3 and variance = 6.
+        kplus_actual = [3745, 933, 230, 0, 0, 0, 0, 0, 0, 0]
+        rp = ReachPoint([5000], kplus_actual)
+        gpm = GammaPoissonModel([rp])
+        Imax, _, _, _ = gpm._fit_histogram_chi2_distance(rp, nstarting_points=1)
+        gpm2 = GammaPoissonModel([rp], extrapolation_multiplier=4.0)
+        Imax2, _, _, _ = gpm2._fit_histogram_chi2_distance(rp, nstarting_points=1)
+        self.assertAlmostEqual(Imax, 16500, delta=100)
+        self.assertAlmostEqual(Imax2, 20000, delta=100)
```

Comment:
1. Wondering what are the source(s) of this delta.  Is `scipy.optimize.minimize` the only source?
2. Does larger extrapolation_multiplier (say, 10) pull Imax closer to 30000?

Reply:
The comment that I gave at the top of this function was not complete.  The input histogram was formed by starting with the histogram of expected values per the model parameters.  Then, I zeroed out all of the values starting with frequency 4.  This simulates the effect of adding differentially private noise to the frequency buckets.  The resulting value for Imax was then 16540, and for Imax2 was 19999.99.  If also tried a run with the extrapolation_multiplier set to 10.  In this case, Imax was returned as 50023.20.  The fact that it does not exactly match the target of 50000 could be due to a couple possible causes: (1) At this value, the relative change in the value of the objective function may be below the tolerance threshold used by the optimizer; or, (2) this represents a local minimum of the objective function.

Reply:
Thanks for the super clear explanations!